packages/compass/src/main/window-manager.ts

```diff
@@ -25,6 +25,31 @@ import {
 
 const debug = createDebug('mongodb-compass:electron:window-manager');
 
+export const EXCLUDED_MONGODB_HOSTS = [
+  'compass-maps.mongodb.com',
+  'evergreen.mongodb.com',
+  'downloads.mongodb.com',
+  'cloud.mongodb.com',
+];
+
+// Exported for testing purposes only
+export const urlWithUtmParams = (urlString: string): string => {
+  try {
+    const url = new URL(urlString);
+    const urlShouldHaveUtmParams =
+      /^(.*\.)?mongodb\.com$/.test(url.hostname) &&
+      !EXCLUDED_MONGODB_HOSTS.includes(url.hostname);
+
+    if (urlShouldHaveUtmParams) {
+      url.searchParams.set('utm_source', 'compass');
+      url.searchParams.set('utm_medium', 'product');
```

Comment:
Here we are overriding the exisiting `utm` params in url. If that's intended, looks good.

Reply:
Yes, that is intended. We came to this conclusion after a small thought work [here](https://github.com/mongodb-js/compass/pull/4446#issuecomment-1569949401). The appropriate condition to not enforce these utm values would have been the case when the code will be shared across other tools but fortunately, with your last feedback, this part of the logic is very compass specific.

Reply:
But we can always revisit this piece if the situation changes in the future.